### ISSUE TITLE
Problem with accuracy after exporting resnet-like model to OpenVino

### ISSUE THREAD

**Issue by korotaS** [openvinotoolkit/nncf#1610]
Hi! 
In our team we've been using NNCF with OpenVino for a while and we've faced the same issue twice (mentioned in #1284 and #1590). In this issue I would like to formalize the problem a bit for more basic models from `torchvision` (we were using customized versions of resnet and regnet). Sorry for a lot of code, but it would be easier to reproduce. 

Suppose we have a resnet-like model (ResNet-50 for example) and a simple task to quantize model and check its performance by measuring accuracy on [imagenette](https://github.com/fastai/imagenette) dataset (not full ImageNet because imagenette is easier to find). Let's initialize dataset, dataloader, model and criterion: 
```python
from copy import deepcopy

from torchvision.models import resnet50, ResNet50_Weights
from torchvision import transforms, datasets
import torch.nn as nn
from nncf import NNCFConfig

device = 'cuda:0'
weights = ResNet50_Weights.DEFAULT

path = 'data/imagenet/imagenette2/val/'
dataset = datasets.ImageFolder(
    root=path,
    transform=weights.transforms()
)
dataloader = torch.utils.data.DataLoader(
    dataset, 
    batch_size=128, 
    num_workers=4, 
    shuffle=False
)
init_dataloader = deepcopy(dataloader)

model = resnet50(weights=weights).to(device)
model.eval()
criterion = nn.CrossEntropyLoss()
criterion = criterion.to(device)
```

Next, let's write our simplest quantization config:
```python
nncf_config = {
    "input_info": {
      "sample_size": [1, 3, 224, 224]
    },
    "compression": {
        "algorithm": "quantization",
    }
}
nncf_config = NNCFConfig.from_dict(nncf_config)
```
And do the quantization:
```python
from nncf.torch.initialization import register_default_init_args
from nncf.torch import create_compressed_model

nncf_config = register_default_init_args(
    nncf_config, init_dataloader, criterion=criterion, criterion_fn=criterion,
    val_loader=dataloader, device=device
)
compression_ctrl, model = create_compressed_model(model, nncf_config, None)
onnx_fpath = f'data/resnet_nncf_issue.onnx'
compression_ctrl.export_model(onnx_fpath, save_format='onnx_13')
```

Now let's introduce methods for inferencing OpenVino and for accuracy calculation. The OpenVino engine I use the same as in #1284 (with one small change - model resize to (-1, 3, -1, -1)).
```python
import numpy as np
from sklearn.metrics import accuracy_score
from tqdm import tqdm

def validate(model, val_loader):
    # mapping between imagenette classes and imagenet classes
    imagenette_mapping = [0, 217, 482, 491, 497, 566, 569, 571, 574, 701]
    
    predictions = []
    references = []
    
    with torch.no_grad():
        for images, target in tqdm(val_loader):
            if isinstance(model, nn.Module):  # original torch model
                output = model(images.to(device)).cpu()
            else:  # nncf openvino model
                output = model(images)
    
            predictions.append(np.argmax(output, axis=1))
            references.append(target)

    predictions = np.concatenate(predictions, axis=0)
    predictions = [imagenette_mapping.index(p) if p in imagenette_mapping else -1 
                   for p in predictions]
    references = np.concatenate(references, axis=0)  

    return accuracy_score(predictions, references)
```
We are ready. Now we create 3 models and calculate their accuracy:
```python
model_original = resnet50(weights=weights).to(device)
model_original.eval();
model_nncf = compression_ctrl._model
model_ov = OpenVinoEngine('data/resnet_nncf_issue.onnx', 4)

validate(model_original, dataloader)  # 0.918
validate(model_nncf, dataloader)  # 0.860
validate(model_ov, dataloader)  # 0.0
```
After export the accuracy drops to 0. But if I change config and ad this line
```python
"ignored_scopes": ["{re}.*/__iadd___*"]  # for regnet it is __add___ instead of __iadd___
```
The values become 
```python
validate(model_original, dataloader)  # 0.918
validate(model_nncf, dataloader)  # 0.862
validate(model_ov, dataloader)  # 0.862
```
I tested this problem on resnet-ish models (ResNet, ResNeXt, RegNet) and it appears everywhere. In MobileNet, for example, there is no such problem. So, as I said in #1284, I think the problem is in skip-connections where `add`/`iadd` operation meets `relu`. 
If this is the case, I think it needs to be written somewhere in the docs or examples (in current version there are no `ignored_scopes` in resnet examples).

**Comment by vshampor** [comment]
@korotaS, thank you for providing detailed reproduction code and information. I will check this more carefully and try to see whether NNCF is at fault here, or whether something can be improved on the NNCF side here.

**Comment by vshampor** [comment]
So, @korotaS, this is what I am getting with the code below (this is basically the same as your code, but I had to use batch size 1 for the OV model because the model couldn't be reshaped otherwise):
```python3
from copy import deepcopy

import onnxruntime
import torch
from torchvision.models import resnet50, ResNet50_Weights
from torchvision import transforms, datasets
import torch.nn as nn
from nncf import NNCFConfig
import openvino.runtime as ov

device = 'cpu'
weights = ResNet50_Weights.DEFAULT

path = 'imagenette2/val/'
dataset = datasets.ImageFolder(
    root=path,
    transform=weights.transforms()
)
dataloader = torch.utils.data.DataLoader(
    dataset,
    batch_size=128,
    num_workers=0,
    shuffle=False
)
init_dataloader = deepcopy(dataloader)

model = resnet50(weights=weights).to(device)
model.eval()
criterion = nn.CrossEntropyLoss()
criterion = criterion.to(device)

nncf_config = {
    "input_info": {
      "sample_size": [1, 3, 224, 224]
    },
    "compression": {
        "algorithm": "quantization",
    }
}
nncf_config = NNCFConfig.from_dict(nncf_config)

from nncf.torch.initialization import register_default_init_args
from nncf.torch import create_compressed_model

nncf_config = register_default_init_args(
    nncf_config, init_dataloader, criterion=criterion, criterion_fn=criterion,
    val_loader=dataloader, device=device
)
compression_ctrl, model = create_compressed_model(model, nncf_config, None)
onnx_fpath = f'resnet_int8.onnx'
compression_ctrl.export_model(onnx_fpath, save_format='onnx_13')

import numpy as np
from sklearn.metrics import accuracy_score
from tqdm import tqdm


def validate(model, val_loader):
    # mapping between imagenette classes and imagenet classes
    imagenette_mapping = [0, 217, 482, 491, 497, 566, 569, 571, 574, 701]

    predictions = []
    references = []

    with torch.no_grad():
        for images, target in tqdm(val_loader):
            if isinstance(model, nn.Module):  # original torch model
                output = model(images.to(device)).cpu()
            elif isinstance(model, onnxruntime.InferenceSession):
                _input_name = model.get_inputs()[0].name
                _output_names = [model.get_outputs()[0].name]
                output = model.run(_output_names, {_input_name: images.numpy()})[0]
            elif isinstance(model, OpenVinoEngine):
                output = model.process(images)
            else:
                # nncf openvino model
                output_layer = model.output(0)
                output = model(images)[output_layer]

            predictions.append(np.argmax(output, axis=1))
            references.append(target)

    predictions = np.concatenate(predictions, axis=0)
    predictions = [imagenette_mapping.index(p) if p in imagenette_mapping else -1
                   for p in predictions]
    references = np.concatenate(references, axis=0)

    return accuracy_score(predictions, references)

model_original = resnet50(weights=weights).to(device)
model_original.eval()
model_nncf = compression_ctrl._model

class OpenVinoEngine:
    def __init__(self, model_fpath, num_workers):
        self.core = ov.Core()
        self.raw_model = self.core.read_model(model_fpath, "AUTO")
        self.raw_model.reshape([1, 3, (1, 2048), (1, 2048)])
        self.model = self.core.compile_model(self.raw_model, "CPU", config={"INFERENCE_NUM_THREADS": str(num_workers)})
        self.infer_request = self.model.create_infer_request()

    def process(self, batch):
        self.infer_request.infer([batch])
        output = [out.data[:] for out in self.infer_request.output_tensors]
        if len(output) > 1:
            return output
        return output[0]


model_ov = OpenVinoEngine(onnx_fpath, 4)

dataloader_one = torch.utils.data.DataLoader(
    dataset,
    batch_size=1,
    num_workers=0,
    shuffle=False
)

print(f"original: {validate(model_original, dataloader)}")  # 0.918
print(f"compressed-torch: {validate(model_nncf, dataloader)}")  # 0.860
print(f"compressed-ov-fq: {validate(model_ov, dataloader_one)}")  # 0.863
```

And the output of the prints is:
```
100%|██████████| 31/31 [06:57<00:00, 13.47s/it]
original: 0.9182165605095541
100%|██████████| 31/31 [21:07<00:00, 40.90s/it]
compressed-torch: 0.8601273885350319
100%|██████████| 3925/3925 [01:16<00:00, 51.63it/s]
compressed-ov-fq: 0.8631847133757962
```

This is the version without ignored_scopes and it seems to be giving good accuracy in OV just fine. I am using `openvino==2022.3.0` and nncf from current develop at `8c39a248a127154ba310dadffab8129a03709d75`.

**Comment by korotaS** [comment]
I see what is happening here... 
I was using OpenVino version 2022.2.0 (for some reason 2022.3.0 was not suitable for me, I don't even remember why, maybe something wasn't working correctly), so I thought it was an NNCF/exporter issue. 
If I switch to OV 2022.3.0, this example works correctly just as in your answer. 
So smth is wrong with OpenVino...

**Comment by MaximProshin** [comment]
@korotaS , as it turned out that the issue was in an older OpenVINO version, I suppose we can close this issue. Any objections?

**Comment by MaximProshin** [comment]
I close it as I think it was resolved. Otherwise, feel free to contact us.
